backend/graminsta/core/services.py

```diff
@@ -0,0 +1,49 @@
+# -*- coding: UTF-8 -*-
+
+"""
+Here defines services provided for views.py
+"""
+
+from .serializers import UserSerializer
+from .serializers import UserInfoSerializer
+from .models import UserInfo
+
+
+def create_or_update_userinfo(validated_data):
```

Comment:
As API level, prefer to separate create and update into two APIs, since the API can evolve into more complex.